File: doc/ex-mathlang-20200821/parsers.js
"use strict";
exports.__esModule = true;
exports.prsr2 = exports.prsr1 = void 0;
function prsr1(str) {
    var tokens = str.split(/\b/);
    console.log(tokens);
    if (tokens.length == 1
        && "0".charCodeAt(0) <= tokens[0].charCodeAt(0)
        && "9".charCodeAt(0) >= tokens[0].charCodeAt(0)) {
        return parseInt(tokens[0]);
    }
    else {
        console.error("expecting a single numeric digit");
    }
}
exports.prsr1 = prsr1;
function prsr2(str) {
    var tokens = str.split(/\b/);
    console.log(tokens);
    if (tokens.length == 1) {
        return parseInt(tokens[0]);
    }
    if ((tokens[0] + tokens[2]).match(/^[0-9]+$/)) {
        if (tokens[1] == "*") {
            return tokens[0] * tokens[2];
        }
        if (tokens[1] == "+") {
            return parseInt(tokens[0]) + parseInt(tokens[2]);
        }
    }
}
exports.prsr2 = prsr2;
console.log(prsr2("2") + " should be 2");
console.log(prsr2("2+2") + " should be 4");
console.log(prsr2("2*4") + " should be 8");

File: doc/ex-mathlang-20200821/parsers.py
#!/usr/bin/python3
import re
def prsr1 (str):
    if len(str) == 1:
       if ord('0') <= ord(str) <= ord('9'):
           return(int(str))
       else:
           print("i can only handle a simple integer right now")
           raise ValueError
    else:
        print("i can only handle a single character right now")
        raise ValueError
print("2 should be 2:", prsr1("2") == 2)

def prsr2 (str):
    tokens = re.split(r'\b',str)
    if len(tokens) == 3:
        return(tokens[1])
    if len(tokens) == 5:
        if (tokens[2] == "*"): return int(tokens[1]) * int(tokens[3])
        if (tokens[2] == "+"): return int(tokens[1]) + int(tokens[3])
print(prsr2("2+2"), "should be 4")
